src/main/java/com/github/havardh/javaflow/phases/verifier/ClassGetterNamingVerifier.java

```diff
@@ -0,0 +1,70 @@
+package com.github.havardh.javaflow.phases.verifier;
+
+import static java.lang.String.format;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.github.havardh.javaflow.ast.Class;
+import com.github.havardh.javaflow.ast.Field;
+import com.github.havardh.javaflow.ast.Method;
+import com.github.havardh.javaflow.ast.Type;
+import com.github.havardh.javaflow.exceptions.AggregatedException;
+import com.github.havardh.javaflow.exceptions.FieldGettersMismatchException;
+
+public class ClassGetterNamingVerifier implements Verifier {
```

Comment:
This class is missing tests.

I wrote the attached proposal for unit tests for this class. **It is missing** one vital test which inspects the exception thrown by the verifier (namely the FieldGettersMismatchException). However there are two failing tests in this suite. Could you have a look at them and see if you agree on the proposed behaviour. 

1) I propose to make the verifier check all field in a class before reporting failures
2) I propose that we should also verify the type of the field and getter. 

```
diff --git a/src/main/java/com/github/havardh/javaflow/ast/builders/MethodBuilder.java b/src/main/java/com/github/havardh/javaflow/ast/builders/MethodBuilder.java
index c862aee..43cca49 100644
--- a/src/main/java/com/github/havardh/javaflow/ast/builders/MethodBuilder.java
+++ b/src/main/java/com/github/havardh/javaflow/ast/builders/MethodBuilder.java
@@ -1,4 +1,34 @@
 package com.github.havardh.javaflow.ast.builders;
 
-public class MethodBuilder {
+import com.github.havardh.javaflow.ast.Method;
+import com.github.havardh.javaflow.ast.Type;
+
+public class MethodBuilder implements Builder<Method> {
+
+  private String name;
+  private Type type;
+
+  public static MethodBuilder methodBuilder() {
+    return new MethodBuilder();
+  }
+
+  public MethodBuilder withName(String name) {
+    this.name = name;
+    return this;
+  }
+
+  public MethodBuilder withType(Type type) {
+    this.type = type;
+    return this;
+  }
+
+  /**
+   * Build a {@code Method} with the builder configuration
+   *
+   * @return the {@code Method}
+   */
+  @Override
+  public Method build() {
+    return new Method(name, type);
+  }
 }
diff --git a/src/test/java/com/github/havardh/javaflow/phases/verifier/ClassGetterNamingVerifierTest.java b/src/test/java/com/github/havardh/javaflow/phases/verifier/ClassGetterNamingVerifierTest.java
index d1a6edb..7bca225 100644
--- a/src/test/java/com/github/havardh/javaflow/phases/verifier/ClassGetterNamingVerifierTest.java
+++ b/src/test/java/com/github/havardh/javaflow/phases/verifier/ClassGetterNamingVerifierTest.java
@@ -1,5 +1,161 @@
-import static org.junit.jupiter.api.Assertions.*;
+package com.github.havardh.javaflow.phases.verifier;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+import com.github.havardh.javaflow.ast.Class;
+import com.github.havardh.javaflow.ast.Type;
+import com.github.havardh.javaflow.ast.builders.ClassBuilder;
+import com.github.havardh.javaflow.ast.builders.FieldBuilder;
+import com.github.havardh.javaflow.ast.builders.MethodBuilder;
+import com.github.havardh.javaflow.ast.builders.TypeBuilder;
+import com.github.havardh.javaflow.exceptions.AggregatedException;
+import com.github.havardh.javaflow.model.CanonicalName;
 
 class ClassGetterNamingVerifierTest {
 
+  private static final Type PRIMITIVE_INT = TypeBuilder.primitive(CanonicalName.primitive("int"));
+  private static final Type PRIMITIVE_DOUBLE = TypeBuilder.primitive(CanonicalName.primitive("double"));
+
+  private ClassGetterNamingVerifier verifier = new ClassGetterNamingVerifier();
+
+  @Test
+  public void shouldPassForModelWithMatchingGetters() {
+    Class aClass = ClassBuilder.classBuilder()
+        .withPackageName("com.github.havardh.models")
+        .withName("Class")
+        .withField(FieldBuilder.fieldBuilder()
+            .withName("someField")
+            .withType(PRIMITIVE_INT)
+            .build())
+        .withGetter(MethodBuilder.methodBuilder()
+            .withName("getSomeField")
+            .withType(PRIMITIVE_INT)
+            .build())
+        .build();
+
+    verifier.verify(singletonList(aClass));
+  }
+
+  @Test
+  public void shouldFailForModelWithGetterWithNonMatchingType() {
+    Class aClass = ClassBuilder.classBuilder()
+        .withPackageName("com.github.havardh.models")
+        .withName("Class")
+        .withField(FieldBuilder.fieldBuilder()
+            .withName("someField")
+            .withType(PRIMITIVE_INT)
+            .build())
+        .withGetter(MethodBuilder.methodBuilder()
+            .withName("getSomeField")
+            .withType(PRIMITIVE_DOUBLE)
+            .build())
+        .build();
+
+    AggregatedException exception = assertThrows(
+        AggregatedException.class,
+        () -> verifier.verify(singletonList(aClass))
+    );
+
+    assertThat(exception.getExceptions(), hasSize(1));
+  }
+
+  @Test
+  public void shouldFailForModelWithGetterWithNonMatchingName() {
+    Class aClass = ClassBuilder.classBuilder()
+        .withPackageName("com.github.havardh.models")
+        .withName("Class")
+        .withField(FieldBuilder.fieldBuilder()
+            .withName("someField")
+            .withType(PRIMITIVE_INT)
+            .build())
+        .withGetter(MethodBuilder.methodBuilder()
+            .withName("getSomeFields")
+            .withType(PRIMITIVE_INT)
+            .build())
+        .build();
+
+    AggregatedException exception = assertThrows(
+        AggregatedException.class,
+        () -> verifier.verify(singletonList(aClass))
+    );
+
+    assertThat(exception.getExceptions(), hasSize(1));
+  }
+
+  @Test
+  public void shouldFailForModelWithoutGetters() {
+    Class aClass = ClassBuilder.classBuilder()
+        .withPackageName("com.github.havardh.models")
+        .withName("Class")
+        .withField(FieldBuilder.fieldBuilder()
+            .withName("someField")
+            .withType(PRIMITIVE_INT)
+            .build())
+        .build();
+
+    AggregatedException exception = assertThrows(
+        AggregatedException.class,
+        () -> verifier.verify(singletonList(aClass))
+    );
+
+    assertThat(exception.getExceptions(), hasSize(1));
+  }
+
+  @Test
+  public void shouldFailAndReportMultipleFieldsPerClass() {
+    Class aClass = ClassBuilder.classBuilder()
+        .withPackageName("com.github.havardh.models")
+        .withName("Class")
+        .withField(FieldBuilder.fieldBuilder()
+            .withName("fieldOne")
+            .withType(PRIMITIVE_INT)
+            .build())
+        .withField(FieldBuilder.fieldBuilder()
+            .withName("fieldTwo")
+            .withType(PRIMITIVE_INT)
+            .build())
+        .build();
+
+    AggregatedException exception = assertThrows(
+        AggregatedException.class,
+        () -> verifier.verify(singletonList(aClass))
+    );
+
+    assertThat(exception.getExceptions(), hasSize(2));
+  }
+
+  @Test
+  public void shouldFailAndReportOnMultipleErroneousClasses() {
+    Class aClass = ClassBuilder.classBuilder()
+        .withPackageName("com.github.havardh.models")
+        .withName("aClass")
+        .withField(FieldBuilder.fieldBuilder()
+            .withName("fieldOne")
+            .withType(PRIMITIVE_INT)
+            .build())
+        .build();
+    Class bClass = ClassBuilder.classBuilder()
+        .withPackageName("com.github.havardh.models")
+        .withName("bClass")
+        .withField(FieldBuilder.fieldBuilder()
+            .withName("fieldOne")
+            .withType(PRIMITIVE_INT)
+            .build())
+        .build();
+
+    AggregatedException exception = assertThrows(
+        AggregatedException.class,
+        () -> verifier.verify(asList(aClass, bClass))
+    );
+
+    assertThat(exception.getExceptions(), hasSize(2));
+  }
+
}
```

Reply:
I've updated tests and added a validation for compatibility of field and getter type.